library/Makefile

```diff
@@ -1,8 +1,21 @@
 
 # Also see "include/mbedtls/config.h"
 
+ifeq ($(CC),iccarm)
+CFLAGS	?= --arm -Om --silent
+
+# gcc comparable warning flags are not present in IAR.
+# All warnings are enabled by default.
+WARNING_CFLAGS ?=
+else
 CFLAGS	?= -O2
 WARNING_CFLAGS ?=  -Wall -W -Wdeclaration-after-statement
+endif
+
+# IAR archive command does not have ar compatible options 
+ifneq ($(AR),iarchive)
```

Comment:
No problem. I will align to mbedtls practice and make it based on env variables.